File: src/main/kotlin/com/github/darderion/mundaneassignmentpolice/checker/rule/Rule.kt
package com.github.darderion.mundaneassignmentpolice.checker.rule

import com.github.darderion.mundaneassignmentpolice.checker.RuleViolation
import com.github.darderion.mundaneassignmentpolice.checker.RuleViolationType
import com.github.darderion.mundaneassignmentpolice.pdfdocument.PDFDocument
import com.github.darderion.mundaneassignmentpolice.pdfdocument.PDFRegion

abstract class Rule(
	val area: PDFRegion,
	val name: String,
	val type: RuleViolationType,
	val description: String
) {
	abstract fun process(document: PDFDocument): List<RuleViolation>
}

File: src/main/webapp/src/classes/Rule.ts
export default class Rule {
    area: string;
    name: string;
    type: string;
    description: string;

    constructor(area: string, name: string, type: string, description: string) {
        this.area = area;
        this.name = name;
        this.type = type;
        this.description = description;
    }
}
